src/rtc.c

```diff
@@ -0,0 +1,518 @@
+ *
+ * @param[out]  calibration
+ *              Destination pointer for the RTC calibration data.
+ */
+void joybus_rtc_read_control( uint16_t * control, uint32_t * calibration )
+{
+    uint64_t data;
+    joybus_rtc_read( 0, &data );
+
+    if( control != NULL ) *control = data >> 48;
+    if( calibration != NULL ) *calibration = data;
+}
+
+/**
+ * @brief Read the current date/time from the real-time clock.
+ *
+ * Your code should call this once per frame to update the rtc_time_t
+ * data structure.
+ *
+ * Before calling this function for the first time, it is recommended
+ * that you call #joybus_rtc_run to ensure the RTC is ready and running.
+ *
+ * The result of calling this function when the RTC is not present is
+ * undefined and may not be safe. Make sure you call #joybus_rtc_status
+ * at the start of your program to ensure the RTC is detected before
+ * reading the current time!
+ *
+ * @param[out]  rtc_time
+ *              Destination pointer for the RTC time data structure
+ */
+void joybus_rtc_read_time( rtc_time_t * rtc_time )
+{
+    uint8_t data[sizeof(uint64_t)];
+    joybus_rtc_read( 2, (uint64_t *)data );
+
+    rtc_time->sec = bcd_to_byte(data[0]);
+    rtc_time->min = bcd_to_byte(data[1]);
+    rtc_time->hour = bcd_to_byte(data[2] - 0x80);
+    rtc_time->day = bcd_to_byte(data[3]);
+    rtc_time->week_day = bcd_to_byte(data[4]);
+    rtc_time->month = bcd_to_byte(data[5]) - 1;
+    rtc_time->year = bcd_to_byte(data[6]);
+    rtc_time->year += (uint16_t)bcd_to_byte(data[7]) * 100;
+    rtc_time->year += 1900;
+}
+
+/**
+ * @brief Write the control block to the real-time clock.
+ *
+ * Typically this would be set to one of two "modes":
+ * Use #JOYBUS_RTC_CONTROL_MODE_SET before writing to block 2 to set the time.
+ * Use #JOYBUS_RTC_CONTROL_MODE_RUN after writing to block 2 to resume the RTC.
+ *
+ * It may take up to 20 milliseconds after this function returns for the
+ * write to actually complete on real hardware. It is recommended to wait
+ * using #JOYBUS_RTC_WRITE_DELAY after setting the RTC control block before
+ * calling #joybus_rtc_read_time or #joybus_rtc_write_time to ensure the
+ * control block data was fully written.
+ *
+ * It is strongly recommended to propagate the calibration data returned from
+ * #joybus_rtc_read_control into the calibration parameter of this function.
+ * If you have any reason to modify or omit the calibration data, please
+ * update this documentation with an explanation of why and how you would
+ * want to do this.
+ *
+ * Generally, you should not need to call this function directly. Prefer
+ * calling the high-level #joybus_rtc_run and #joybus_rtc_set functions which
+ * handle the delays and calibration propagation.
+ *
+ * @param[in]  control
+ *             The control block data
+ *
+ * @param[in]  calibration
+ *             The opaque calibration data from #joybus_rtc_read_control
+ */
+void joybus_rtc_write_control( uint16_t control, uint32_t calibration )
```

Comment:
Static

Reply:
Addressed in https://github.com/DragonMinded/libdragon/pull/152/commits/e46fbb07a6cbb01a98f05a2d3b170d4ff1860b77